src/core/sm/launcher/launcher.cpp

```diff
@@ -63,6 +63,12 @@ Error Launcher::Start()
         return AOS_ERROR_WRAP(err);
     }
 
+    mIsRunning = true;
+
+    if (auto err = mRebootThread.Run([this](void*) { RunRebootThread(); }); !err.IsNone()) {
```

Comment:
You don't need separate thread for reboot handling. It can be handled in FinishLaunch for example.